crates/jstzd/src/task/octez_node.rs

```diff
@@ -0,0 +1,264 @@
+use super::Task;
+use anyhow::Result;
+use async_dropper::AsyncDrop;
+use async_trait::async_trait;
+use std::{fs::File, path::PathBuf, sync::Arc};
+use tokio::{runtime::Handle, sync::RwLock};
+
+#[cfg(test)]
+use tests::{Child, InnerOctezNode};
+
+#[cfg(not(test))]
+use octez::OctezNode as InnerOctezNode;
+#[cfg(not(test))]
+use std::process::Child;
+
+pub struct OctezNodeConfig {
+    binary_path: PathBuf,
+    data_dir: PathBuf,
+    network: String,
+    http_endpoint: String,
+    rpc_endpoint: String,
+    num_connections: u32,
+    log_file: PathBuf,
+    options: Vec<String>,
+}
```

Comment:
Some of these options can have sensible defaults. Additionally I think it might be clearer to use a builder pattern to construct the config, as `OctezNodeConfig::new` (without an LSP i.e. in GitHub review) is rather difficult to read / parse

Reply:
I've added one config builder. `options` is left as a vector for now. That will be addressed in a separate PR along with updating the underlying async octez node interface so that now I don't need to convert options into strings.

Reply:
Changes LGTM